source/simulator/introspection.cc

```diff
@@ -64,8 +64,12 @@ namespace aspect
       b.temperature = fevs.variable("temperature").block_index;
 
       unsigned int n_compositional_fields = fevs.variable("compositions").n_components();
+      const unsigned int first_composition_block_index = fevs.variable("compositions").block_index;
       for (unsigned int i=0; i<n_compositional_fields; ++i)
-        b.compositional_fields.push_back(fevs.variable("compositions").block_index+i);
+        {
+          b.compositional_fields.push_back(fevs.variable("compositions").block_index+i);
+          b.compositional_field_sparsity_pattern.push_back(first_composition_block_index);
```

Comment:
This will have to change when you allow for different kinds of elements, right? In other words, this is the only part that you will have to touch later on?

Reply:
Correct. We will later need to assign them to the first compositional field of the same degree/type.